state/helpers_test.go

```diff
@@ -61,7 +63,8 @@ func makeAndApplyGoodBlock(state sm.State, height int64, lastCommit *types.Commi
 	if err := blockExec.ValidateBlock(state, block); err != nil {
 		return state, types.BlockID{}, err
 	}
-	blockID := types.BlockID{Hash: block.Hash(), PartsHeader: types.PartSetHeader{}}
+	blockID := types.BlockID{Hash: block.Hash(),
+		PartsHeader: types.PartSetHeader{Total: 3, Hash: cmn.RandBytes(32)}}
```

Comment:
For my own education: Can you tell me what the PartsHeader is, and why we're setting the Hash now when we weren't before?

Reply:
https://github.com/tendermint/tendermint/blob/master/docs/spec/blockchain/blockchain.md#blockid

It's basically an alternative hash for the block that's only used for gossipping the block during consensus but has unfortunately leaked into the more global BlockID. A better design might see it removed from the BlockID, but would probably make the consensus reactor more complicated. Probably it's filled in now because there's more validation and its expected to be populated

Reply:
Correct. It's because I've added a panic in Vote#CommitSig() func if BlockID is neither Empty nor Complete. Normally ValidateBasic() catches this kind of errors, but since in tests we construct objects manually you will get a panic if PartsHeader is empty here